crates/types/src/block_impl.rs

```diff
@@ -45,21 +70,6 @@ impl Committable for VIDTransaction {
 
 impl Transaction for VIDTransaction {}
 
-/// The error type for block payload.
-#[derive(Snafu, Debug)]
-pub enum BlockPayloadError {
-    /// Previous state commitment does not match
-    PreviousStateMismatch,
-    /// Nonce was reused
-    ReusedTxn,
-    /// Genesis failure
-    GenesisFailed,
-    /// Genesis reencountered after initialization
-    GenesisAfterStart,
-    /// invalid block
-    InvalidBlock,
-}
-
 /// A [`BlockPayload`] that contains a list of `VIDTransaction`.
 #[derive(PartialEq, Eq, Hash, Serialize, Deserialize, Clone, Debug)]
 pub struct VIDBlockPayload {
```

Comment:
I'm a little confused about this `VIDBlockPayload` struct. My understanding is that any impl of `BlockPayload` inside hotshot is purely for example or testing purposes. So why does this file `block_impl.rs` even exist in production code? Where is it used?

Reply:
@jbearer why does `block_impl.rs` need to exist? Any production code implementing `BlockPayload` should not be in hotshot, correct?

Reply:
All the implementations in `block_impl.rs` are for examples/tests only. Ideally, it should be moved to a place explicitly for non-production code, but files that depend on it are not well-organized, so moving it may cause dependency issues. I'll create an issue to organize our non-production code.